doit.go

```diff
@@ -139,6 +139,37 @@ func (v Version) Complete(lv LatestVersioner) string {
 	return buffer.String()
 }
 
+// CompleteJSON is the complete version for doit, formatted as JSON.
+func (v Version) CompleteJSON(lv LatestVersioner) string {
+	versionInfo := &struct {
+		Version      string `json:"version,omitempty"`
+		Commit       string `json:"commit,omitempty"`
+		Notification string `json:"notification,omitempty"`
+	}{
+		Version: fmt.Sprintf("doctl version %s", v.String()),
+	}
+
+	if v.Build != "" {
+		versionInfo.Commit = fmt.Sprintf("\nGit commit hash: %s", v.Build)
+	}
+
+	if tagName, err := lv.LatestVersion(); err == nil {
+		v0, err1 := semver.Make(tagName)
+		v1, err2 := semver.Make(v.String())
+
+		if len(v0.Build) == 0 {
+			v0, err1 = semver.Make(tagName + "-release")
+		}
+
+		if err1 == nil && err2 == nil && v0.GT(v1) {
+			versionInfo.Notification = fmt.Sprintf("release %s is available, check it out!", tagName)
```

Comment:
should this be `tagName` ? 

I think we should eliminate the need to do any string parsing 100% for json.